spicy/toolchain/src/compiler/codegen/parser-builder.cc

```diff
@@ -163,6 +163,26 @@ struct ProductionVisitor : public production::Visitor {
                               hilti::statement::comment::Separator::After);
     }
 
+    void _checkSizeAmount(hilti::Attribute* size, Expression* ncur, type::unit::item::Field* field = nullptr) {
+        // Make sure we parsed the entire &size amount.
+        auto missing =
+            builder()->lower(builder()->memberCall(state().cur, "offset"), builder()->memberCall(ncur, "offset"));
+        auto insufficient = builder()->addIf(missing);
+        pushBuilder(insufficient, [&]() {
+            if ( field && ! field->isAnonymous() )
+                // Clear the field in case the type parsing has started
+                // to fill it.
+                builder()->addExpression(builder()->unset(state().self, field->id()));
+
+            auto want = *size->valueAsExpression();
+            auto got = builder()->difference(builder()->memberCall(state().cur, "offset"),
+                                             builder()->grouping(
+                                                 builder()->difference(builder()->memberCall(ncur, "offset"), want)));
+            pb->parseError("&size amount not consumed: expected %" PRIu64 ", but got %" PRIu64, {want, got},
```

Comment:
Let's add a unit to `want` and `got`, I believe `bytes` (opposed to say elements).